compiler/rustc_parse/src/parser/expr.rs

```diff
@@ -2820,6 +2820,15 @@ impl<'a> Parser<'a> {
                 }
                 self.recover_struct_comma_after_dotdot(exp_span);
                 break;
+            } else if self.token == token::DotDotDot && self.look_ahead(1, |t| t.is_ident()) {
+                // suggest `..v` instead of `...v`
+                self.sess.emit_err(MissingDotDot {
+                    token_span: self.token.span,
+                    sugg_span: self.token.span,
+                });
+                self.bump();
+                self.bump(); // bump the next ident
```

Comment:
This probably doesn't handle something like `...Default::default()` correctly -- can you use a snapshot and `parse_expr` instead?

Reply:
Great point, updated the code.